dlt/common/configuration/specs/azure_credentials.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional, Dict, Any
+
+from dlt.common import pendulum
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.specs import CredentialsConfiguration, CredentialsWithDefault, configspec
+from dlt.common.configuration.specs.exceptions import InvalidBoto3Session
+from dlt import version
+
+import fsspec
+
+
+@configspec
+class AzureCredentialsWithoutDefaults(CredentialsConfiguration):
+    """Credentials for azure blob storage, compatible with adlfs"""
+
+    azure_storage_account_name: str = None
+    azure_storage_account_key: Optional[TSecretStrValue] = None
+    azure_storage_sas_token: TSecretStrValue = None
+
+    def to_adlfs_credentials(self) -> Dict[str, Any]:
+        """Return a dict that can be passed as kwargs to adlfs"""
+        return dict(
+            account_name=self.azure_storage_account_name,
+            account_key=self.azure_storage_account_key,
+            sas_token=self.azure_storage_sas_token,
+        )
+
+    def create_sas_token(self) -> None:
+        from azure.storage.blob import generate_account_sas, ResourceTypes
+        self.azure_storage_sas_token = generate_account_sas(  # type: ignore[assignment]
+            account_name=self.azure_storage_account_name,
+            account_key=self.azure_storage_account_key,
+            resource_types=ResourceTypes(container=True, object=True),
+            permission='rwdlacx',
```

Comment:
yeah you are probably right to do it after all. please add permissions to to configuration. default should be r/w like here. maybe we do not need execution ;>

we'll be implementing fsspec source and there we could use r/o token

Reply:
Added the permission setting.  `racwdl` I think allows everything we need